### PR TITLE
feat: support update blocking for EOL systems

### PR DESCRIPTION
**Pull request by flying-sausages**
## Fixes issues: 
- #395 

## Proposed Changes:
- Deprecate `xenial`
- Move git update part of `box` to `script/update/10-git.sh`
- Add logic to choose swizzin branch/tag to check out based on `_os_codename`

## Categories

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update
- Some existing `functions` got changed (e.g. `utils`, `os`, `apt`, `ask`, etc.)
- Changes structure of project (major rewrites, large directory changes, etc.)
- Impacts the flow of how we contribute (IDE changes, `.github` dir, etc.)
(Yes, all apply)

## Checklist

- [x] Docs have been made OR are not necessary
    - PR link: https://github.com/liaralabs/docs.swizzin.ltd/pull/44
- [x] Changes to panel have been made OR are not necessary
    - PR link: 
- [x] Code is formatted [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Shellcheck isn't screaming [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Prints to terminal are handled [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#printing-into-the-terminal)
- [x] I have commented my code, particularly in hard-to-understand areas

## Testing done
OS & Version: Xenial VM

### How have you tested this
- This has been tested on a Xenial VM and functions as intended

### CODE REVIEW

**Review by brettpetch — COMMENTED** [review]
Sausage, this commit looks good. I'll run some testing to make sure everything is in order if you haven't already. Lets get this v3 bread!

/b

**Comment by brettpetch** [comment]
Issue here: 

`error: unknown switch `C'`

**Comment by liaralabs** [comment]
If you are renaming the dependencies updater, this change needs to be reflected in `setup.sh` as well. (Sorry I can't comment directly on a file rename)

File: scripts/box [hunk 1]
```diff
@@ -124,28 +124,9 @@ function _clr() {
...
-        git reset --hard origin/master
-        echo_info "HEAD is now set to $(git log --pretty=format:'%h' -n1)"
-        echo_progress_done "Local repository updated"
-    fi
-
-    # source globals again in case changes were made
-    #shellcheck source=sources/globals.sh
-    . /etc/swizzin/sources/globals.sh
-
+    bash /etc/swizzin/sources/repo.sh || exit 1
```
Comment (liaralabs) [inline scripts/box:127]: Line 127 and Line 140: One of these is not like the other, which is it?

File: sources/repo.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        echo_error "Failed to update from git"
+        exit 1
+    }
+    echo_progress_done "Local repository updated from ${SWIZ_GIT_REF:-master}"
+    echo_info "HEAD is now set to $(git log --pretty=format:'%h' -n1)"
+fi
+export SWIZ_REPO_SCRIPT_RAN=true
+# source globals again in case changes were made
+#shellcheck source=sources/globals.sh
+. /etc/swizzin/sources/globals.sh
```
Comment (liaralabs) [inline sources/repo.sh:40]: This won't work here.

Comment (flying-sausages) [reply sources/repo.sh:40]: ah duh you're right.

**Comment by flying-sausages** [comment]
tag moved here

**Pull request merged**